staging/src/k8s.io/client-go/tools/cache/mutation_cache.go

```diff
@@ -128,6 +140,32 @@ func (c *mutationCache) ByIndex(name string, indexKey string) ([]interface{}, er
 			items = append(items, obj)
 		}
 	}
+
+	if c.includeAdds {
+		fn := c.indexer.GetIndexers()[name]
+		// Keys() is returned oldest to newest, so full traversal does not alter the LRU behavior
+		for _, key := range c.mutationCache.Keys() {
+			if keySet.Has(key.(string)) {
+				continue
+			}
+			updated, ok := c.mutationCache.Get(key)
+			if !ok {
+				continue
+			}
+			elements, err := fn(updated)
+			if err != nil {
+				return nil, err
```

Comment:
I would log and continue.  I think this is a nice to have, not a "fail me" layer.